Classes/Provider/AbstractProvider.php

```diff
@@ -464,11 +464,30 @@ protected function getCurrentValuePointerName() {
 	 */
 	protected function getPageValues() {
 		$record = $GLOBALS['TSFE']->page;
+		$restoreVars = array();
 		if ($GLOBALS['TSFE']->sys_language_uid != 0) {
-			$localisation = $this->recordService->get('pages_language_overlay', '*', 'pid = "' . $record['uid'] . '" AND sys_language_uid = "' . $GLOBALS['TSFE']->sys_language_uid . '"');
+			if (isset($record['uid']) && isset($record['pid'])) {
+				$restoreVars[] = array(
+					'uid' => $record['uid'],
+					'pid' => $record['pid']
+				);
+			}
+			$localisation = $this->recordService->get(
+				'pages_language_overlay',
+				'*',
+				'pid = "' . $record['uid'] .
+				'" AND sys_language_uid = "' . $GLOBALS['TSFE']->sys_language_uid . '"' .
```

Comment:
We should have this extended query for sure - but the $restoreVars approach I'm not too sure about. It should probably not be an array-within-an-array, and I'd instead define a new $mergedRecord:

```php
$mergedRecord = RecursiveArrayUtility::merge($record, $localisation);
$mergedRecord['uid'] = $record['uid'];
$mergedRecord['pid'] = $record['pid'];
```

alternatively:

```php
unset($localisation['uid'], $localisation['pid']);
$mergedRecord = RecursiveArrayUtility::merge($record, $localisation);
```

It would read better and we probably don't expect to be adding additional properties to this backup/restore behavior.